### PR TITLE
Add `VideoReaderDecoder` test for missing labels

### PR DESCRIPTION
**Pull request by awolant**
## Category:

Refactoring

## Description:

Extracted common parts of `VideoLoaderDecoderCpu` and `VideoLoaderDecoderGpu` to new base class `VideoLoaderDecoderBase`

Added new test for `VideoReaderDecoderCpu` that triggers assertion present in this new base class.

### Affected modules and functionalities:

`VideoLoaderDecoderCpu` and `VideoLoaderDecoderGpu`, video tests

### Tests
- [ ] Existing tests apply
- [x] New tests added
  - [ ] Python tests
  - [x] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [x] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [x] Affects existing requirements
- [ ] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: dali/operators/reader/loader/video/frames_decoder_test.cc [hunk 1]
```diff
@@ -92,17 +92,6 @@ class FramesDecoderTestBase : public VideoTestBase {
...
-    try {
-      body();
-
-      FAIL();   // If we reached this point body did not throw an exception.
-    } catch (const DALIException &e) {
-      EXPECT_TRUE(
-        strstr(e.what(), expected_error.c_str()));
-    }
-  }
-
```
Comment (awolant) [inline dali/operators/reader/loader/video/frames_decoder_test.cc:105]: This was moved to `VideoTestBase`, so it can be reused in other video tests.

File: dali/test/python/test_dali_cpu_only.py [hunk 2]
```diff
@@ -46,7 +46,9 @@
 coco_dir = os.path.join(data_root, 'db', 'coco', 'images')
 coco_annotation = os.path.join(data_root, 'db', 'coco', 'instances.json')
 sequence_dir = os.path.join(data_root, 'db', 'sequence', 'frames')
-video_files = get_files(os.path.join('db', 'video', 'vfr'), 'mp4')
+video_files = [
+    os.path.join(get_dali_extra_path(), 'db', 'video', 'vfr', 'test_1.mp4'),
+    os.path.join(get_dali_extra_path(), 'db', 'video', 'vfr', 'test_2.mp4')]
```
Comment (awolant) [inline dali/test/python/test_dali_cpu_only.py:51]: Before we were listing all files in the directory. We are adding new files to this directory in other PRs. This caused this test case to get bigger due to some unrelated changes. I decided to change this behavior.

File: dali/operators/reader/video_reader_decoder_op_test.cc [hunk 3]
```diff
@@ -225,6 +225,14 @@ TEST_F(VideoReaderDecoderCpuTest, VariableFrameRate_CpuOnlyTests) {
...
 }
 
+TEST_F(VideoReaderDecoderCpuTest, LabelMismatch_CpuOnlyTests) {
+  std::vector<std::string> paths {cfr_hevc_videos_paths_[0]};
+
+  RunFailureTest([&]() -> void {
+    RunTest<dali::CPUBackend>(paths, cfr_videos_);},
+    "Current pipeline object is no longer valid.");
+}
+
```
Comment (awolant) [inline dali/operators/reader/video_reader_decoder_op_test.cc:235]: New test. We have one video file and two labels. Assertion in the operator construction should be triggered.

**Review by JanuszL — APPROVED** [review]

File: dali/operators/reader/loader/video/video_test_base.cc [hunk 4]
```diff
@@ -144,4 +144,15 @@ void VideoTestBase::SaveFrame(
...
+  try {
+    body();
+
+    FAIL();   // If we reached this point body did not throw an exception.
+  } catch (const std::exception &e) {
+    EXPECT_TRUE(
+      strstr(e.what(), expected_error.c_str()));
+  }
+}
+
```
Comment (awolant) [inline dali/operators/reader/loader/video/video_test_base.cc:157]: This was moved from FramesDecoders tests, so it can be reused in more video tests.

File: dali/test/python/test_dali_cpu_only.py [hunk 5]
```diff
@@ -46,7 +46,9 @@
 coco_dir = os.path.join(data_root, 'db', 'coco', 'images')
 coco_annotation = os.path.join(data_root, 'db', 'coco', 'instances.json')
 sequence_dir = os.path.join(data_root, 'db', 'sequence', 'frames')
-video_files = get_files(os.path.join('db', 'video', 'vfr'), 'mp4')
+video_files = [
```
[the remark this answers is not in the document]
Comment (awolant) [reply dali/test/python/test_dali_cpu_only.py:49]: Done

**Review by JanuszL — APPROVED** [review]

File: dali/operators/reader/loader/video/video_loader_decoder_base.h [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  explicit inline VideoLoaderDecoderBase(const OpSpec &spec):
+    filenames_(spec.GetRepeatedArgument<std::string>("filenames")),
+    sequence_len_(spec.GetArgument<int>("sequence_length")),
+    stride_(spec.GetArgument<int>("stride")),
+    step_(spec.GetArgument<int>("step")) {
+    has_labels_ = spec.TryGetRepeatedArgument(labels_, "labels");
+    DALI_ENFORCE(
+        !has_labels_ || labels_.size() == filenames_.size(),
+        make_string(
+            "Number of provided files and lables should match. Provided ",
```
Comment (jantonguirao) [inline dali/operators/reader/loader/video/video_loader_decoder_base.h:52]:
```suggestion
            "Number of provided files and labels should match. Provided ",
```

**Review by jantonguirao — APPROVED** [review]

File: dali/operators/reader/loader/video/video_loader_decoder_base.h (hunk 6 above)
Comment (awolant) [reply dali/operators/reader/loader/video/video_loader_decoder_base.h]: Done

**Comment by awolant** [comment]
!build
